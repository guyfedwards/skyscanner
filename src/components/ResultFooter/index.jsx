import React from 'react';
import propTypes from 'prop-types';
import styled from 'styled-components';
import BpkButton from 'bpk-component-button';
import BpkText from 'bpk-component-text';

const Root = styled.div`
  display: flex;
  justify-content: space-between;

  margin-top: 10px;
`;

const Agent = styled(BpkText)`
  color: #aba9a9;
  line-height: 1rem;
`;

const Left = styled.div`
  display: flex;
  flex-direction: column;
`;

const Right = styled.div`
  display: flex;
  align-items: flex-end;
`;

const ResultFooter = ({ agent, price }) => (
  <Root>
    <Left>
      <BpkText textStyle="xxl">{price}</BpkText>
      <Agent textStyle="lg">{agent}</Agent>
    </Left>
    <Right>
      <BpkButton large>Select</BpkButton>
    </Right>
  </Root>
);

ResultFooter.propTypes = {
  agent: propTypes.string.isRequired,
  price: propTypes.string.isRequired,
};

export default ResultFooter;
