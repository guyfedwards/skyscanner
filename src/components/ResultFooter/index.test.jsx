import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import ResultFooter from '.';

describe('ResultFooter', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<ResultFooter agent="CheapFlights" price="£35" />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ResultFooter agent="CheapFlights" price="£35" />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
