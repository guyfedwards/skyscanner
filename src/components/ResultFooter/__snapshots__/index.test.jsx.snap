// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultFooter should render correctly 1`] = `
<ForwardRef(styled.div)>
  <ForwardRef(styled.div)>
    <BpkText
      bold={false}
      className={null}
      tagName="span"
      textStyle="xxl"
    >
      Â£35
    </BpkText>
    <ForwardRef(Styled(BpkText))
      textStyle="lg"
    >
      CheapFlights
    </ForwardRef(Styled(BpkText))>
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)>
    <BpkButton
      blank={false}
      className={null}
      destructive={false}
      disabled={false}
      featured={false}
      href={null}
      iconOnly={false}
      large={true}
      link={false}
      onClick={null}
      outline={false}
      rel={null}
      secondary={false}
      submit={false}
    >
      Select
    </BpkButton>
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;
